### ISSUE TITLE
FK Fields Unexpectedly Reordered During Model Structure Fix

### ISSUE THREAD

**Issue by pbryant-phew** [pgmodeler/pgmodeler#1705]
**Bug description**
When moving from 0.9.3 to 0.9.4, and following the instructions to apply non-optional "Fix Model Structure", the fix runs OK (at least in this case) but foreign key (FK) fields/columns become re-ordered.  This causes problems both when there is a preferred order for FK columns (ie where the field order helps confirm the purpose and intentions of this table), and _particularly_ when there are eg defined functions that rely on ROW() statements with row fields expected to be in a particular order.  This makes it practically impossible to upgrade even one revision level of pgModeler (or to the upcoming 1.0.0+) without embarking on a rather large project of un-re-ordering fields across hundreds of tables (eg 334 tables currently, or which almost all of them have FK fields).

**How to reproduce**
Have a model with FK references between tables.
Move from (eg) 0.9.3 to 0.9.4.
Accept non-optional "fix" procedure in order to open the model in 0.9.4.
Observe that FK fields have been re-ordered (eg to the end of the column list for all relevant tables)

**Expected behavior**
FK fields are left in the location they were specified to be in the model.

**Screenshots**
_If applicable, add screenshots to help explain your problem._

**Info about your desktop**
 - OS: macOS
 - Version: 12.5
 - pgModeler version: 0.9.4

**Comment by rkhaotix** [comment]
Hey @pbryant-phew, 

Unfortunately, collateral effects like this can happen. That's why I need such feedback from the users. It's almost impossible to catch all cases no matter how I test everything carefully.

Now, despite changing the order of the columns, have the FK semantics changed? For example, in your original model, you had an FK(col1, col2, col3) and after applying the fix does this same FK has turned into FK(col2, col1, col3) or something like this? 

I know that the order is important for you, but for now, my concern is if after the fix the FKs changed their structure which is not desired. I can review the fix procedures to try to keep the original columns ordering.

**Comment by pbryant-phew** [comment]
Yes, completely understand, and great to be able to raise and describe these here for everyone's benefit.

As far as I can tell so far, the only problem is order of the auto-generated FK column in the foreign key table.

**Comment by rkhaotix** [comment]
> As far as I can tell so far, the only problem is order of the auto-generated FK column in the foreign key table.

That's great, then! You gave me some helpful info... I will focus on the resolution of this for 1.0.0 too! Thanks :)
